core/app/app.go

```diff
@@ -0,0 +1,213 @@
+package app
```

Comment:
All the app and connector stuff now lives here. This "app" abstraction feels great.